### DISCUSSION TITLE
How to search/sort by foreign key column?

### DISCUSSION THREAD

**Discussion by anton-petrov** [aminalaee/sqladmin#514] [Q&A] [answered]
Hi! I have a table with foreign key column, which points to another table (entity). So, I see that column in general list, and it's ok, but including this column in `column_sortable_list` or `column_searchable_list` causes such an error:
```python
sqlalchemy.exc.CompileError: Can't resolve label reference for ORDER BY / GROUP BY / DISTINCT etc.
```
So, the question is how to sort/search by foreign object, for example, by it's name? I alreade have defined __str__ method, but it doesn't help.

Thanks for advance.

**Comment by anton-petrov** [comment]
```python
class DataSources(Base):
source_id = Column(SmallInteger)
    ...
    def __str__(self):
        return self.data_source_name if self.data_source_name else str(self.data_source_id)

class Devices(Base):
    ...
    data_source_name = Column(Text, nullable=False)
    data_source = relationship("DataSources", back_populates="devices")
```

**Reply by aminalaee** [reply]
Do you mean sort by ForeignKey column or a related model's column?
 I think looks like you want to sort for example By `Devices.data_source.source_id` which doesn't work right now.

**Reply by anton-petrov** [reply]
Yes, exactly.

**Reply by aminalaee** [reply, marked as answer]
Yeah thats missing for now, I will however take a look
